src/Sema.zig

```diff
@@ -12470,11 +12470,171 @@ fn zirReify(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.I
             const ty = try Type.Tag.error_set_merged.create(sema.arena, names);
             return sema.addType(ty);
         },
-        .Enum => return sema.fail(block, src, "TODO: Sema.zirReify for Enum", .{}),
+        .Enum => {
+            const struct_val = union_val.val.castTag(.@"struct").?.data;
+            // TODO use reflection instead of magic numbers here
+            // error_set: type,
+            // layout: ContainerLayout,
+            const layout_val = struct_val[0];
+            // tag_type: type,
+            const tag_type_val = struct_val[1];
+            // fields: []const EnumField,
+            const fields_val = struct_val[2];
+            // decls: []const Declaration,
+            const decls_val = struct_val[3];
+            // is_exhaustive: bool,
+            const is_exhaustive_val = struct_val[4];
+
+            // enum layout is always auto
+            const layout = layout_val.toEnum(std.builtin.Type.ContainerLayout);
+            if (layout != .Auto) {
+                return sema.fail(block, src, "reified enums must have a layout .Auto", .{});
+            }
+
+            // Decls
+            const decls_slice_val = decls_val.castTag(.slice).?.data;
+            const decls_decl = decls_slice_val.ptr.pointerDecl().?;
+            try sema.ensureDeclAnalyzed(decls_decl);
```

Comment:
this is a no-op because creating an anonymous decl creates it with already `complete` status.